### ISSUE TITLE
Why is it that `result.insertId` can be number or string when calling `insert()`

### ISSUE THREAD

**Issue by mateuscb** [sidorares/node-mysql2#636]
The [code](https://github.com/sidorares/node-mysql2/blob/f5eabbe7f329117baac8dd121b15bdc383a01b71/lib/packets/packet.js#L202) and these tests: ([number](https://github.com/sidorares/node-mysql2/blob/5f0fb8f1f5035e2c0207490aa2f0b838dc82fdc2/test/integration/connection/test-insert-bigint-big-number-strings.js#L27), [string](https://github.com/sidorares/node-mysql2/blob/5f0fb8f1f5035e2c0207490aa2f0b838dc82fdc2/test/integration/connection/test-insert-bigint-big-number-strings.js#L34)) make me think that `result.insertId` can be either a number or a string. If that is correct, what was the reasoning for doing that? It puts the burden on the caller to deal with both types.

If it is expected behavior, I think it would be nice to update the documentation to state that it could return a number or a string.

I have enabled both big number configs:  `{supportBigNumbers: true, bigNumberStrings: true}`

**Comment by sidorares** [comment]
I would expect it to follow same rules as other numbers - number if can fit 53 bit, string if bigNumberStrings and supportBigNumbers

ref #108

**Comment by sidorares** [comment]
main reasoning is that mysql type accuracy can be such that it cannot fit into JS number.

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER

**Comment by mateuscb** [comment]
Just to clarify: 
if it can fit in 53bit it will always be number, regardless if `bigNumberStrings` and `supportBigNumbers` are set.
If it can't fit in 53bit. it will return string if `bigNumberStrings` and `supportBigNumbers` are set, and error if they are not set?

Thanks for the quick response!

**Comment by mateuscb** [comment]
Just saw your last comment. I get the accuracy limitation, my doubt is why not always return a string?
That way the caller doesn't have to worry about it being a string or a number.

**Comment by sidorares** [comment]
yes, I kind of agree with you, I prefer to treat insertId as non-numeric unique id. Right now thie behaviour is made to be consistent with normal rows result where most of the time one prefer to have numbers as numbers. If we decide to change insertId  behaviour that would require major version bump

**Comment by pbomb** [comment]
Doesn't the insertId behavior _not_ follow the same rules? When the new row id is small, insertId gives me back a number, but when I `select` from that same row, the id comes back as a string. This is with `bigNumberStrings` and `supportBigNumbers` enabled. I would expect the id to be treated consistently, regardless of whether it came from an `insert` or `select` and consider this behavior to not be self-consistent.

**Comment by sidorares** [comment]
@pbomb I'm pretty sure the intention was to have this consistent, reopening to check and fix if not

**Comment by dvisztempacct** [comment]
Will there be any movement on this issue? Thanks

**Comment by sidorares** [comment]
@dvisztempacct no movement, thank for bringing attention, would like to fix that

( any help to fix would be appreciated as well - in any form, even pr with failing test covering this behaviour )

**Comment by godlanbo** [comment]
So is there any updated progress to solve this problem?
